tests/Functional/AutoincrementColumnTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+
+use function strtolower;
+
+class AutoincrementColumnTest extends FunctionalTestCase
+{
+    /** @var bool */
+    private static $tableCreated = false;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (! self::$tableCreated) {
+            $table = new Table('auto_increment_table');
+            $table->addColumn('id', 'integer', ['autoincrement' => true]);
+            $table->addColumn('test_int', 'integer');
+            $table->setPrimaryKey(['id']);
+
+            $sm = $this->connection->getSchemaManager();
+            $sm->createTable($table);
+            self::$tableCreated = true;
+        }
+
+        $platform = $this->connection->getDatabasePlatform();
+
+        $this->connection->executeStatement(
+            $platform->getTruncateTableSQL('auto_increment_table')
+        );
+    }
+
+    public function testInsertArbitraryId(): void
+    {
+        if ($this->connection->getDatabasePlatform()->getName() === 'mssql') {
+            self::markTestSkipped('Test does not work on MSSql.');
+        }
+
+        $this->connection->insert('auto_increment_table', ['test_int' => 1]);
+
+        $this->assertEquals([
+            ['id' => 1, 'test_int' => 1],
+        ], $this->lowerCaseResultColumnNames($this->connection->fetchAllAssociative(
+            'SELECT * FROM auto_increment_table ORDER BY id ASC'
+        )));
+
+        $this->connection->insert('auto_increment_table', ['id' => 1000, 'test_int' => 2]);
```

Comment:
Why is the rest of the test needed? The above assertion verifies that the resulting value is the same as specified manually (could use a different value than `1` though).

Reply:
IMHO we also need to test what happens next time a record is inserted _after_ an arbitrary value is inserted. This tests in Oracle, for instance, the rest of the trigger code that aligns the sequence value with the max value inserted,

Reply:
Why does it matter what happens if it's inconsistent across the platforms? The user cannot rely in this behavior anyways.

What does matter though is what happens if we create a new table, and then:
1. Insert an empty row (`1`)
2. Insert `2` manually. At this point, MySQL will bump `AUTO_INCREMENT` to `3` but Oracle will just skip the call to `seq.NEXTVAL`
3. Insert an empty row.

It doesn't matter what the resulting value is since it's inconsistent across platforms. It does matter whether it succeeds or fails.

Reply:
That scenario rules off the test for DB2 and PostgreSql, because the manual insertion in 2 does not update their sequence so 3 fails because it will try to insert a value '2' in the ID again.